From a184231c5fd6fa58c6a6c05bb4d0e0f17e05115d Mon Sep 17 00:00:00 2001
From: Sveinung Kvilhaugsvik <sveinung84@users.sourceforge.net>
Date: Tue, 13 Jun 2017 16:00:13 +0200
Subject: [PATCH 2/2] Inform foreign unit receiver.

A unit can be transferred to a foreign nation by setting its home city to
a city owned by another player. Let the receiver know about the unit he was
given.

See hrm Feature #676658
---
 server/unithand.c | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/server/unithand.c b/server/unithand.c
index 7ce2e13..e28c58c 100644
--- a/server/unithand.c
+++ b/server/unithand.c
@@ -2680,8 +2680,26 @@ void unit_change_homecity_handling(struct unit *punit, struct city *new_pcity,
 static bool do_unit_change_homecity(struct unit *punit,
                                     struct city *pcity)
 {
+  const char *giver = NULL;
+
+  if (unit_owner(punit) != city_owner(pcity)) {
+    /* This is a gift. Tell the receiver. */
+    giver = player_name(unit_owner(punit));
+  }
+
   unit_change_homecity_handling(punit, pcity, TRUE);
 
+  if (punit->homecity == pcity->id && giver) {
+    /* Notify the city owner about the gift he received. */
+    notify_player(city_owner(pcity), city_tile(pcity), E_UNIT_BUILT,
+                  ftc_server,
+                  /* TRANS: other player ... unit type ... city name. */
+                  _("%s transferred control over a %s to you in %s."),
+                  giver,
+                  unit_tile_link(punit),
+                  city_link(pcity));;
+  }
+
   return punit->homecity == pcity->id;
 }
 
-- 
2.1.4

